boring/src/ssl/callbacks.rs

```diff
@@ -95,73 +103,90 @@ where
         + Sync
         + Send,
 {
-    unsafe {
-        let ssl = SslRef::from_ptr_mut(ssl);
-        let callback_idx = SslContext::cached_ex_index::<F>();
-
-        let callback = ssl
-            .ssl_context()
-            .ex_data(callback_idx)
-            .expect("BUG: psk callback missing") as *const F;
-        let identity = if identity.is_null() {
-            None
-        } else {
-            Some(CStr::from_ptr(identity).to_bytes())
-        };
-        // Give the callback mutable slices into which it can write the psk.
-        let psk_sl = slice::from_raw_parts_mut(psk as *mut u8, max_psk_len as usize);
-        match (*callback)(ssl, identity, psk_sl) {
-            Ok(psk_len) => psk_len as u32,
-            Err(e) => {
-                e.put();
-                0
-            }
+    // SAFETY: boring provides valid inputs.
+
+    let ssl = unsafe { SslRef::from_ptr_mut(ssl) };
+
+    let identity = if !identity.is_null() {
```

Comment:
nit: it's better avoid reverse conditions as it's easy to misread them or miss the negation during refactoring. So, let's return to the previous variant

Reply:
Ah yeah I copy-pasted that from `raw_client_psk` and then renamed `hint` to `authority`, I'll return to previous way here and change `raw_client_psk`.